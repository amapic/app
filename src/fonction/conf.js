 const adresse="http://localhost"
 export default adresse;