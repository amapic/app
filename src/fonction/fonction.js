function createData(time, amount,amount2) {
    return {
        time,
        amount,
        amount2
    };
}

export function setDataLim(data){
console.log(data);
   console.log(data.jour);
    console.log(data.timestamp);

}

export async function getdata(url) {
    const response = await fetch(url)
    const responseData = await response.json()
    console.log(response)
    console.log(responseData)
    if (response.ok) {
        console.log("ok");
        var data = [];
    var arrayTime = [];
    var arrayAmount = [];
        
        for (let key in responseData.datetime){
            let value = responseData.datetime[key];
            arrayTime.push(value);
        }
        for (let key in responseData.n_cum_dose1){
            let value = responseData.n_cum_dose1[key];
            arrayAmount.push(value);
        }
         arrayAmount.forEach((x, i) => data.push(createData(arrayTime[i].toString(), i,i/2)));
         var ff=data.slice(1,10);
         return ff
    } else {
        alert(JSON.stringify(responseData))
        return {responseData, loading: false}
    }
}