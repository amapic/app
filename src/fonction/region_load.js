import DATA from '../data/france_geojson.js';
var _ = require('underscore');


var list_poly = []

DATA.features.forEach(function (item) {
	
  var Coords = item.geometry.coordinates;

  var dimensions = [
	Coords.length,
	Coords.reduce((x, y) => Math.max(x, y.length), 0)
  ];


  var paths_list = [];
  // var X=0;
  for (let pas = 0; pas <= dimensions[0] - 1; pas++) {

	var X;
	if (dimensions[0]===1){
		X=[[Coords][pas][0]];
	}else if (item.properties.code==="84"){
		X=[[Coords][0][pas]];
	}else{
		X=[Coords][0][pas];
	}

	var paths = _.map([
	 X // doit Ãªtre un array avec une valeur
	], function (entry) {
	  return _.reduce(entry, function (list, polygon) {
		// This map() only transforms the data.
		_.each(_.map(polygon, function (point) {
		  return {"lat":point[1],"lng": point[0]};
		}), function (point) {
		  list.push(point);
		});

		return list;
	  }, []);
	});
	// console.log(paths.length);
	paths_list.push(paths[0]);
  }
  
	  if (paths_list.length === 1) {
		list_poly.push(paths_list[0]);
	  }else{
		  list_poly.push(paths_list[0]);
	  }

});

export default list_poly