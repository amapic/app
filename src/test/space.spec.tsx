import { chercheData } from '../fonction/fonction'
import React, { useState } from 'react';
import Lien from '../component/navbar/navbar'
import renderer from 'react-test-renderer';
import adresse from '../fonction/conf'



const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('Space test suite', () => {
  it('My Space Test', () => {
    expect(true).toEqual(true);
  });


  // it('tests /destinations endpoints', async () => {

  //     // expect(["tt"]).toBeInstanceOf(Array)
  //     // expect(chercheData()).toMatch(/.*/)
  //     let hh=await chercheData()

  //     expect(hh).toBeInstanceOf(Object)
  //     expect(hh[0]).toBeInstanceOf(Object)
  //     expect(Object.keys(hh).length).toBeGreaterThan(500)

  // });
  
  // it('renders correctly lien vers page vaccin', () => {
  //   useRouter.mockImplementationOnce(() => ({
  //     pathname: '/planet'
  //   }))
  //   const tree = renderer
  //     .create(<Lien />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // it('renders correctly lien vers page planète', () => {
  //   useRouter.mockImplementationOnce(() => ({
  //     pathname: '/planets'
  //   }))
  //   const tree = renderer
  //     .create(<Lien />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
  it('adresse origine de l api', () => {
    
    expect(adresse).toEqual("http://68.183.74.150")
  });

  it('api_planete', () => {
    let hh = chercheData().then(planet=>{
      expect(planet).toBeDefined()
      expect(planet[0].star_name).toBeDefined()
      expect(Object.keys(planet).length).toEqual(4872)
    })
  });

  it('test graph type de planète', () => {
    const tree = renderer
      .create(<Lien />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });



});