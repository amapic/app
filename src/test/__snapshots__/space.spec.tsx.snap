// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space test suite renders correctly 1`] = `
<span
  className="navtext"
  onClick={[Function]}
>
   ◄  Dashboard exoplanète
</span>
`;
